### PR TITLE
[Fix](Outfile) Use data_type_serde to export data to csv file format

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

1. Modify the outfile logic, use the `data type serde` framework.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_hll_serde.cpp [hunk 1]
```diff
@@ -37,21 +37,31 @@ namespace doris {
...
 }
 
-void DataTypeHLLSerDe::serialize_one_cell_to_json(const IColumn& column, int row_num,
-                                                  BufferWritable& bw,
-                                                  FormatOptions& options) const {
+Status DataTypeHLLSerDe::serialize_one_cell_to_json(const IColumn& column, int row_num,
+                                                    BufferWritable& bw, FormatOptions& options,
+                                                    int nesting_level) const {
+    if (!options._output_object_data) {
+        if (nesting_level >= 2) {
```
Comment (morningman) [inline be/src/vec/data_types/serde/data_type_hll_serde.cpp:50]: Add some comment for this `if`

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — DISMISSED** [review]

**Review by AshinGau — DISMISSED** [review]

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Comment by BePPPower** [comment]
run p1

**Review by kaka11chen — APPROVED** [review]
LGTM.

**Review by zy-kkk — APPROVED** [review]
LGTM

**Pull request merged**